### DISCUSSION TITLE
Mission Manager instance not being created

### DISCUSSION THREAD

**Discussion by Pranayomg** [SimDynamX/SpaceTeamsPRO#56] [2024 Lunar SAR Competition Q&A] [answered]
![image](https://github.com/user-attachments/assets/0eaa3d2f-8c4a-4dd5-aeb6-dc9590b13c56)

I have renamed all the source paths in the sim config file to match the Repo folder name, and I have even tried deleting the repo and reinstalling the entire program but this error 10061 does not seem to go away.
The API link (https://github.com/SimDynamX/SpaceTeamsPro/wiki/STU-Lunar-SAR-API-Docs) does not seem to have any other suggestions other than renaming the source path within the sim config file.

I have attached the log file with the errors, and the Unreal Engine error:
[Sim_STU2_SAR_Task1.log](https://github.com/user-attachments/files/17593401/Sim_STU2_SAR_Task1.log)
![image](https://github.com/user-attachments/assets/3d4af9fd-12f3-4b4d-b199-4a195a3a1733)

Any other ideas or assistance would be greatly appreciated!

**Comment by Bajwa1595** [comment]
Im getting the same error. The other Demos work for me but they doesnt have the necessary Moon rovers configured.

**Reply by connorjak** [reply]
Ok that's a good data point. We may be able to make some testing steps to narrow down the issue.

**Comment by connorjak** [comment]
Relevant portion of the log for reference

```log
2024-10-31T20:53:37 [INFO]----[Main]: Python System Instance MissionManager starting script at Source filepath: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\ExampleMM_TaskGraphSimple.py
2024-10-31T20:53:37 [INFO]----[Main]: "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Versions\v0.27.3\py\Scripts\python.exe" "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\ExampleMM_TaskGraphSimple.py" --sim=Local/Repos/STU_Lunar_SAR_LunarMavericks/sims/STU2_SAR_Task1.json --name=MissionManager --port=41192 --netid=2 --serv_netid=1
2024-10-31T20:53:37 [ERROR]----[Main]: Python System Instance MissionManager Failed to be Created: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\ExampleMM_TaskGraphSimple.py
2024-10-31T20:53:37 [INFO]----[Main]: Python System Instance EntityBehavior_LTV1 starting script at Source filepath: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py
2024-10-31T20:53:37 [INFO]----[Main]: "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Versions\v0.27.3\py\Scripts\python.exe" "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py" --sim=Local/Repos/STU_Lunar_SAR_LunarMavericks/sims/STU2_SAR_Task1.json --name=EntityBehavior_LTV1 --port=41193 --netid=3 --serv_netid=1
2024-10-31T20:53:37 [ERROR]----[Main]: Python System Instance EntityBehavior_LTV1 Failed to be Created: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py
2024-10-31T20:53:37 [INFO]----[Main]: Python System Instance EntityBehavior_LTV2 starting script at Source filepath: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py
2024-10-31T20:53:37 [INFO]----[Main]: "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Versions\v0.27.3\py\Scripts\python.exe" "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py" --sim=Local/Repos/STU_Lunar_SAR_LunarMavericks/sims/STU2_SAR_Task1.json --name=EntityBehavior_LTV2 --port=41194 --netid=4 --serv_netid=1
2024-10-31T20:53:37 [ERROR]----[Main]: Python System Instance EntityBehavior_LTV2 Failed to be Created: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py
2024-10-31T20:53:37 [INFO]----[Main]: Python System Instance ParamWrangler set to Active: false; not initializing it.
2024-10-31T20:53:37 [INFO]----[Main]: Python System Instance CompetitionBackend starting script at Source filepath: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\CompetitionBackend.py
2024-10-31T20:53:37 [INFO]----[Main]: "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Versions\v0.27.3\py\Scripts\python.exe" "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\CompetitionBackend.py" --sim=Local/Repos/STU_Lunar_SAR_LunarMavericks/sims/STU2_SAR_Task1.json --name=CompetitionBackend --port=41195 --netid=5 --serv_netid=1
2024-10-31T20:53:37 [ERROR]----[Main]: Python System Instance CompetitionBackend Failed to be Created: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\CompetitionBackend.py
2024-10-31T20:53:37 [FATAL]----[Main]: System Loading errors!
```
I'm not exactly sure what's happening here, this could happen for a few reasons:
- Not having run FirstTimeSetup.bat, causing the python virtual environment for ST to not be created
- An error during FirstTimeSetup.bat
- Some other issue with the python environment on your computer that conflicts with the one ST tries to set up? Please try `py --list` in the terminal so I can see your list of python versions, and describe what other python installs you know you have on your computer
- ~~Some sort of typo in the paths~~

**Reply by Pranayomg** [reply]
> Relevant portion of the log for reference
> 
> ```
> 2024-10-31T20:53:37 [INFO]----[Main]: Python System Instance MissionManager starting script at Source filepath: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\ExampleMM_TaskGraphSimple.py
> 2024-10-31T20:53:37 [INFO]----[Main]: "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Versions\v0.27.3\py\Scripts\python.exe" "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\ExampleMM_TaskGraphSimple.py" --sim=Local/Repos/STU_Lunar_SAR_LunarMavericks/sims/STU2_SAR_Task1.json --name=MissionManager --port=41192 --netid=2 --serv_netid=1
> 2024-10-31T20:53:37 [ERROR]----[Main]: Python System Instance MissionManager Failed to be Created: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\ExampleMM_TaskGraphSimple.py
> 2024-10-31T20:53:37 [INFO]----[Main]: Python System Instance EntityBehavior_LTV1 starting script at Source filepath: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py
> 2024-10-31T20:53:37 [INFO]----[Main]: "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Versions\v0.27.3\py\Scripts\python.exe" "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py" --sim=Local/Repos/STU_Lunar_SAR_LunarMavericks/sims/STU2_SAR_Task1.json --name=EntityBehavior_LTV1 --port=41193 --netid=3 --serv_netid=1
> 2024-10-31T20:53:37 [ERROR]----[Main]: Python System Instance EntityBehavior_LTV1 Failed to be Created: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py
> 2024-10-31T20:53:37 [INFO]----[Main]: Python System Instance EntityBehavior_LTV2 starting script at Source filepath: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py
> 2024-10-31T20:53:37 [INFO]----[Main]: "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Versions\v0.27.3\py\Scripts\python.exe" "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py" --sim=Local/Repos/STU_Lunar_SAR_LunarMavericks/sims/STU2_SAR_Task1.json --name=EntityBehavior_LTV2 --port=41194 --netid=4 --serv_netid=1
> 2024-10-31T20:53:37 [ERROR]----[Main]: Python System Instance EntityBehavior_LTV2 Failed to be Created: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py
> 2024-10-31T20:53:37 [INFO]----[Main]: Python System Instance ParamWrangler set to Active: false; not initializing it.
> 2024-10-31T20:53:37 [INFO]----[Main]: Python System Instance CompetitionBackend starting script at Source filepath: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\CompetitionBackend.py
> 2024-10-31T20:53:37 [INFO]----[Main]: "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Versions\v0.27.3\py\Scripts\python.exe" "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\CompetitionBackend.py" --sim=Local/Repos/STU_Lunar_SAR_LunarMavericks/sims/STU2_SAR_Task1.json --name=CompetitionBackend --port=41195 --netid=5 --serv_netid=1
> 2024-10-31T20:53:37 [ERROR]----[Main]: Python System Instance CompetitionBackend Failed to be Created: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\CompetitionBackend.py
> 2024-10-31T20:53:37 [FATAL]----[Main]: System Loading errors!
> ```
> 
> I'm not exactly sure what's happening here, this could happen for a few reasons:
> 
> * Not having run FirstTimeSetup.bat, causing the python virtual environment for ST to not be created
> * An error during FirstTimeSetup.bat
> * Some other issue with the python environment on your computer that conflicts with the one ST tries to set up? Please try `py --list` in the terminal so I can see your list of python versions, and describe what other python installs you know you have on your computer
> * Some sort of typo in the paths

I have only Python 3.12 installed.
![image](https://github.com/user-attachments/assets/4e12bbfa-e014-4fe6-aec3-37ad961c3935)
I also attached an image of the "py" folder in the SimDynamX directory, but I feel like there should be more files after running FirstTimeSetup.bat:
![image](https://github.com/user-attachments/assets/57a01005-cb05-4a7a-911b-e268eac11ab7)

**Reply by thecamo6** [reply]
I am having the same issue. The scripts folder of my python is completely missing. Throughout the setup, it says to make you you have python installed as said in the setup, but I cannot find any of that anywhere on how to do this. I have python on my computer, its just messed up inside the program itself.

**Reply by bishnukatuwal** [reply]
@thecamo6 Yes, we had the same problem, but I had a teammate that had those files, and I told him to email me the versions file then I replaced my file with his versions file, and it worked lol.

**Reply by thecamo6** [reply]
I have no such teammate. Can you send them to me or something? I don't know what else to do. my email is camerondivin@gmail.com

**Reply by bishnukatuwal** [reply]
I got you

**Reply by bishnukatuwal** [reply]
But you might run into a problem with the file name being mine, but after you select the interperter on this folder, you can just change back to your systems python file and it should work

**Reply by bishnukatuwal** [reply]
It's sent

**Reply by connorjak** [reply]
> > Relevant portion of the log for reference
> > ```
> > 2024-10-31T20:53:37 [INFO]----[Main]: Python System Instance MissionManager starting script at Source filepath: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\ExampleMM_TaskGraphSimple.py
> > 2024-10-31T20:53:37 [INFO]----[Main]: "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Versions\v0.27.3\py\Scripts\python.exe" "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\ExampleMM_TaskGraphSimple.py" --sim=Local/Repos/STU_Lunar_SAR_LunarMavericks/sims/STU2_SAR_Task1.json --name=MissionManager --port=41192 --netid=2 --serv_netid=1
> > 2024-10-31T20:53:37 [ERROR]----[Main]: Python System Instance MissionManager Failed to be Created: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\ExampleMM_TaskGraphSimple.py
> > 2024-10-31T20:53:37 [INFO]----[Main]: Python System Instance EntityBehavior_LTV1 starting script at Source filepath: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py
> > 2024-10-31T20:53:37 [INFO]----[Main]: "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Versions\v0.27.3\py\Scripts\python.exe" "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py" --sim=Local/Repos/STU_Lunar_SAR_LunarMavericks/sims/STU2_SAR_Task1.json --name=EntityBehavior_LTV1 --port=41193 --netid=3 --serv_netid=1
> > 2024-10-31T20:53:37 [ERROR]----[Main]: Python System Instance EntityBehavior_LTV1 Failed to be Created: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py
> > 2024-10-31T20:53:37 [INFO]----[Main]: Python System Instance EntityBehavior_LTV2 starting script at Source filepath: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py
> > 2024-10-31T20:53:37 [INFO]----[Main]: "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Versions\v0.27.3\py\Scripts\python.exe" "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py" --sim=Local/Repos/STU_Lunar_SAR_LunarMavericks/sims/STU2_SAR_Task1.json --name=EntityBehavior_LTV2 --port=41194 --netid=4 --serv_netid=1
> > 2024-10-31T20:53:37 [ERROR]----[Main]: Python System Instance EntityBehavior_LTV2 Failed to be Created: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\EntityBehavior_LTV.py
> > 2024-10-31T20:53:37 [INFO]----[Main]: Python System Instance ParamWrangler set to Active: false; not initializing it.
> > 2024-10-31T20:53:37 [INFO]----[Main]: Python System Instance CompetitionBackend starting script at Source filepath: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\CompetitionBackend.py
> > 2024-10-31T20:53:37 [INFO]----[Main]: "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Versions\v0.27.3\py\Scripts\python.exe" "C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\CompetitionBackend.py" --sim=Local/Repos/STU_Lunar_SAR_LunarMavericks/sims/STU2_SAR_Task1.json --name=CompetitionBackend --port=41195 --netid=5 --serv_netid=1
> > 2024-10-31T20:53:37 [ERROR]----[Main]: Python System Instance CompetitionBackend Failed to be Created: C:\Users\jayar\AppData\Local\SimDynamX\STPro\Assets\Repos\STU_Lunar_SAR_LunarMavericks\CompetitionBackend.py
> > 2024-10-31T20:53:37 [FATAL]----[Main]: System Loading errors!
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I'm not exactly sure what's happening here, this could happen for a few reasons:
> > 
> > * Not having run FirstTimeSetup.bat, causing the python virtual environment for ST to not be created
> > * An error during FirstTimeSetup.bat
> > * Some other issue with the python environment on your computer that conflicts with the one ST tries to set up? Please try `py --list` in the terminal so I can see your list of python versions, and describe what other python installs you know you have on your computer
> > * Some sort of typo in the paths
> 
> I have only Python 3.12 installed. ![image](https://private-user-images.githubusercontent.com/186496665/382171623-4e12bbfa-e014-4fe6-aec3-37ad961c3935.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA0Mzg0NzMsIm5iZiI6MTczMDQzODE3MywicGF0aCI6Ii8xODY0OTY2NjUvMzgyMTcxNjIzLTRlMTJiYmZhLWUwMTQtNGZlNi1hZWMzLTM3YWQ5NjFjMzkzNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwMVQwNTE2MTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lYjk0MWRkNzcxODk1YTY5ZjBkMzUyZDE4YzIzMDJmZmZhZDM5MTdjM2EzNzBkMzkyM2RiOTNkNDFhNzJjZjU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.u0n5wVKJKhUeflx-z7ibCGhoR2LwqZf16dmKGwwzrvM) I also attached an image of the "py" folder in the SimDynamX directory, but I feel like there should be more files after running FirstTimeSetup.bat: ![image](https://private-user-images.githubusercontent.com/186496665/382171948-57a01005-cb05-4a7a-911b-e268eac11ab7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA0Mzg0NzMsIm5iZiI6MTczMDQzODE3MywicGF0aCI6Ii8xODY0OTY2NjUvMzgyMTcxOTQ4LTU3YTAxMDA1LWNiMDUtNGE3YS05MTFiLWUyNjhlYWMxMWFiNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwMVQwNTE2MTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00MTZiNWRkNmYzYjlmOTVjNjI0YzhhMjFjZTQ2ZjdjZWNmMTUxNDdkYzA5ZmZkNmJhNTZiYjYzNjE5NDgyNGRkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.C8FvCBd35EDqCrLUWisKc529QjJC2yxIsQZ3-NDKTz4)

If you only have Python 3.12 installed, either you didn't run `FirstTimeSetup.bat` or something went wrong when trying to run that script.

The workaround to copy someone else's STPro/Versions/... folder might partially or fully work for now; as far as I remember, there is nothing user-specific about those files.

**Comment by connorjak** [comment, marked as answer]
Seems to be a duplicate of https://github.com/SimDynamX/SpaceTeamsPro/issues/61; this error also happens when python is not installed correctly for the STPRO version that was installed.
